pkg/metrics/timer/timer.go

```diff
@@ -0,0 +1,71 @@
+package timer
+
+import (
+	"strconv"
+	"time"
+
+	"github.com/sirupsen/logrus"
+)
+
+// Timer is the struct that keeps track of each of the sections.
+type Timer struct {
+	startTime    time.Time
+	totalTime    time.Time
+	listOfStages []string
+	stageTimes   map[string]time.Duration
+}
+
+const (
+	// InfrastructureStageName is a constant string value that is used to display the Infrastructure created stage text in the logs.
+	InfrastructureStageName = "Infrastructure"
+	// APIStageName is a constant string value that is used to display the API stage text in the logs.
+	APIStageName = "API"
+	// BootstrapCompleteStageName is a constant string value that is used to display the Bootstrap Complete text in the logs.
+	BootstrapCompleteStageName = "Bootstrap Complete"
+	// BootstrapDestroyStageName is a constant string value that is used to display the Boostrap Destroy Complete text in the logs.
+	BootstrapDestroyStageName = "Bootstrap Destroy"
+	// ClusterOperatorStageName is a constant string value that is used to display the Cluster Operator Complete text in the logs.
+	ClusterOperatorStageName = "Cluster Operator"
+
+	// ConsoleStageName is a constant string value that is used to display the Cluster Operator Complete text in the logs.
+	ConsoleStageName = "Console"
+)
+
+var timer Timer
+
+// StartTimer initiailzes the timer object with the current timestamp information.
+func StartTimer() {
+	timer = Timer{}
+	timer.startTime = time.Now().Round(time.Second)
+	timer.totalTime = time.Now().Round(time.Second)
+	timer.listOfStages = []string{}
+	timer.stageTimes = make(map[string]time.Duration)
+}
+
+// SetTimeElapsed returns the duration of the current stage.
+func SetTimeElapsed(key string) {
+	duration := time.Now().Round(time.Second).Sub(timer.startTime)
+	timer.listOfStages = append(timer.listOfStages, key)
+	timer.stageTimes[key] = duration
+	timer.startTime = time.Now().Round(time.Second)
+}
+
+// LogSummary prints the summary of all the times collected so far into the INFO section.
+func LogSummary() {
+	indent := ""
+	maxLen := 0
+	for _, item := range timer.listOfStages {
+		if len(item) > maxLen {
+			maxLen = len(item)
+		}
+	}
+
+	if maxLen != 0 {
+		logrus.Debugf("Time elapsed for each section")
+	}
+
+	for _, item := range timer.listOfStages {
+		logrus.Debugf(indent+"%-"+strconv.Itoa(maxLen)+"s  :  %v", item, timer.stageTimes[item])
```

Comment:
I think the colon should be "attached" to the left hand word, not in its own column, but that might just be my taste.

Reply:
I'm not sure? but I usually leave a space between everything possible so I feel it looks neat. I think it's a preference? gofmt did not complain.